include/seastar/core/future.hh

```diff
@@ -1918,13 +1926,15 @@ inline
 future<T> make_exception_future(std::exception_ptr&& ex) noexcept {
     return future<T>(exception_future_marker(), std::move(ex));
 }
+SEASTAR_MODULE_EXPORT_END
 
 template <typename T>
 inline
 future<T> internal::make_exception_future(future_state_base&& state) noexcept {
     return future<T>(exception_future_marker(), std::move(state));
 }
 
+SEASTAR_MODULE_EXPORT_BEGIN
```

Comment:
I'm trying to understand the logic of these _END/_BEGINs. Is it to avoid exporting something in the internal namespace? I think we can export it even though it's internal.

Reply:
Or is it something to do with `inline`?

Reply:
> Is it to avoid exporting something in the internal namespace? I think we can export it even though it's internal.

yes, i really want to get it right.